adc/consumer.py

```diff
@@ -84,6 +85,12 @@ def mark_done(self, msg: confluent_kafka.Message, asynchronous: bool = True):
         else:
             self._consumer.commit(msg, asynchronous=False)
 
+    def stop(self):
+        """Stops the runloop of the consumer. Useful when running the
+        consumer in a different thread.
+        """
+        self.running = False
```

Comment:
With the suggestion to use an `Event` above, this would become a call to `set` on that object (whose name should perhaps be inverted, e.g. `self._stop` or `self._stopping`).